python/sanlock.c

```diff
@@ -902,39 +923,39 @@ py_get_lockspaces(PyObject *self __unused, PyObject *args, PyObject *keywds)
             goto exit_fail;
 
         /* fill the dictionary information: lockspace */
-        if ((ls_value = PyString_FromString(lss[i].name)) == NULL)
+        if ((ls_value = PyBytes_FromString(lss[i].name)) == NULL)
             goto exit_fail;
         rv = PyDict_SetItemString(ls_entry, "lockspace", ls_value);
         Py_DECREF(ls_value);
         if (rv != 0)
             goto exit_fail;
 
         /* fill the dictionary information: host_id */
-        if ((ls_value = PyInt_FromLong(lss[i].host_id)) == NULL)
+        if ((ls_value = PyLong_FromLong(lss[i].host_id)) == NULL)
             goto exit_fail;
         rv = PyDict_SetItemString(ls_entry, "host_id", ls_value);
         Py_DECREF(ls_value);
         if (rv != 0)
             goto exit_fail;
 
         /* fill the dictionary information: path */
-        if ((ls_value = PyString_FromString(lss[i].host_id_disk.path)) == NULL)
+        if ((ls_value = PyUnicode_FromString(lss[i].host_id_disk.path)) == NULL)
```

Comment:
Handling paths should be in a separate patch.

How can we create unicode if we don't know the encoding of the path? We cannot assume that the default encoding will work, since in python 2 it is ascii, and in python 3 utf-8.

We must use here the filesystemencoding - you used it in the python FSConverter.